### ISSUE TITLE
Optional dependency tests failing after pymc 5.7.0 release

### ISSUE THREAD

**Issue by oscarbenjamin** [sympy/sympy#25455]
I've opened an issue with pymc:
https://github.com/pymc-devs/pymc/issues/6840

Pymc put out a new release (5.7.0) several hours ago. The optional dependency tests now fail with:
```
$ pytest sympy/stats/sampling/tests/test_sample_finite_rv.py -k test_sample_pymc --tb=short
======================================== test session starts ========================================
platform linux -- Python 3.11.3, pytest-7.3.1, pluggy-1.0.0
architecture: 64-bit
cache:        yes
ground types: gmpy 2.1.5

rootdir: /home/oscar/current/active/sympy
configfile: pyproject.toml
plugins: cov-4.1.0, anyio-3.6.2, xdist-3.2.1, instafail-0.5.0
collected 5 items / 4 deselected / 1 selected                                                       

sympy/stats/sampling/tests/test_sample_finite_rv.py F                                         [100%]

============================================= FAILURES ==============================================
_________________________________________ test_sample_pymc __________________________________________
sympy/stats/sampling/tests/test_sample_finite_rv.py:73: in test_sample_pymc
    samps = sample(X, size=size, library='pymc')
sympy/stats/rv.py:1182: in sample
    return next(iterator)
sympy/stats/rv.py:1352: in return_generator_finite
    d = ps.sample(size=(numsamples,) + ((size,) if isinstance(size, int) else size),
sympy/stats/frv.py:355: in sample
    return {self.value: self.distribution.sample(size, library, seed)}
sympy/stats/rv.py:1639: in sample
    samps = samps.reshape(size)
E   ValueError: cannot reshape array of size 1003 into shape (1,3)
----------------------------------------- Captured log call -----------------------------------------
WARNING  pytensor.tensor.blas:blas_headers.py:1016 Using NumPy C-API based implementation for BLAS functions.
                                          DO *NOT* COMMIT!                                          
====================================== short test summary info ======================================
FAILED sympy/stats/sampling/tests/test_sample_finite_rv.py::test_sample_pymc - ValueError: cannot reshape array of size 1003 into shape (1,3)
================================== 1 failed, 4 deselected in 4.39s ==================================
```
I'm pretty sure that's a pymc bug.

**Comment by oscarbenjamin** [comment]
I can also see some warnings in the optional dependency tests:
```
=============================== warnings summary ===============================
sympy/printing/tests/test_aesaracode.py:27
  /home/runner/work/sympy/sympy/sympy/printing/tests/test_aesaracode.py:27: DeprecationWarning: `Scalar` is deprecated; use `ScalarType` instead.
    from aesara.scalar.basic import Scalar

sympy/utilities/tests/test_lambdify.py::test_issue_20070
  /home/runner/work/sympy/sympy/sympy/utilities/tests/test_lambdify.py:1415: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
    assert numba.jit(f)(1)==0.8414709848078965

sympy/plotting/tests/test_plot.py::test_plot_and_save_5
  /opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/numpy/lib/function_base.py:2415: RuntimeWarning: overflow encountered in cast
    res = asanyarray(outputs, dtype=otypes[0])

sympy/plotting/tests/test_plot_implicit.py::test_region_and
  /home/runner/work/sympy/sympy/sympy/plotting/plot.py:1600: RuntimeWarning: More than 20 figures have been opened. Figures created through the pyplot interface (`matplotlib.pyplot.figure`) are retained until explicitly closed and may consume too much memory. (To control this warning, see the rcParam `figure.max_open_warning`). Consider using `matplotlib.pyplot.close()`.
    self.fig = self.plt.figure(figsize=self.size)

sympy/parsing/tests/test_c_parser.py::test_while_stmt
  /home/runner/work/sympy/sympy/sympy/codegen/ast.py:211: SymPyDeprecationWarning: 
  
  The string fallback in sympify() is deprecated.
  
  To explicitly convert the string form of an object, use
  sympify(str(obj)). To add define sympify behavior on custom
  objects, use sympy.core.sympify.converter or define obj._sympy_
  (see the sympify() docstring).
  
  sympify() performed the string fallback resulting in the following string:
  
  '<built-in function sum>'
  
  See https://docs.sympy.org/latest/explanation/active-deprecations.html#deprecated-sympify-string-fallback
  for details.
  
  This has been deprecated since SymPy version 1.6. It
  will be removed in a future version of SymPy.
  
    return cls._get_constructor(attr)(arg)

sympy/stats/sampling/tests/test_sample_continuous_rv.py::test_sample_pymc
  /opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/multipledispatch/dispatcher.py:27: AmbiguityWarning: 
  Ambiguities exist in dispatched function _unify
  
  The following signatures may result in ambiguous behavior:
  	[object, ConstrainedVar, Mapping], [ConstrainedVar, Var, Mapping]
  	[object, ConstrainedVar, Mapping], [ConstrainedVar, object, Mapping]
  	[object, ConstrainedVar, Mapping], [ConstrainedVar, object, Mapping]
  	[object, ConstrainedVar, Mapping], [ConstrainedVar, Var, Mapping]
  
  
  Consider making the following additions:
  
  @dispatch(ConstrainedVar, ConstrainedVar, Mapping)
  def _unify(...)
  
  @dispatch(ConstrainedVar, ConstrainedVar, Mapping)
  def _unify(...)
  
  @dispatch(ConstrainedVar, ConstrainedVar, Mapping)
  def _unify(...)
  
  @dispatch(ConstrainedVar, ConstrainedVar, Mapping)
  def _unify(...)
    warn(warning_text(dispatcher.name, ambiguities), AmbiguityWarning)

sympy/printing/tests/test_aesaracode.py: 154 warnings
  /opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/site-packages/aesara/tensor/type.py:95: DeprecationWarning: The `broadcastable` keyword is deprecated; use `shape`.
    warnings.warn(

sympy/printing/tests/test_aesaracode.py::test_Derivative
sympy/printing/tests/test_aesaracode.py::test_Derivative
  /home/runner/work/sympy/sympy/sympy/printing/tests/test_aesaracode.py:98: DeprecationWarning: `GraphRewriter.optimize` is deprecated; use `GraphRewriter.rewrite` instead.
    mode.optimizer.optimize(fgraph)
```

**Comment by oscarbenjamin** [comment]
This is fixed in PyMC: https://github.com/pymc-devs/pymc/pull/6843

I don't know if/when they will issue a bug fix release. The optional dependency tests will pass when that happens.

**Comment by oscarbenjamin** [comment]
Looks like this is fixed as of pymc 5.7.1 which was released a few days ago.
